src/BenchmarksClient/Startup.cs

```diff
@@ -177,21 +157,19 @@ public static int Main(string[] args)
                     }
                     else if (job.State == ClientState.Deleting)
                     {
-                        Log($"Deleting job {jobLogText}");
-
-                        Debug.Assert(process != null);
+                        Debug.Assert(worker != null);
+                        Log($"Deleting job {worker.JobLogText}");
 
                         try
                         {
-                            if (!process.HasExited)
-                            {
-                                process.Kill();
-                            }
+                            // Wait 5 seconds before killing worker
+                            var stopTask = worker.StopAsync();
+                            await Task.WhenAny(stopTask, Task.Delay((int)TimeSpan.FromSeconds(5).TotalMilliseconds));
```

Comment:
We should really kill the worker. So `StopAsync` should itself be responsible for killing the process if it's not stopped after some delay. So no need to wait for anything else than `StopAsync`